src/typeDefs/mutations.ts

```diff
@@ -117,21 +117,16 @@ export const mutations = gql`
     ): Plugin!
 
     createAdvertisement(
-      orgId: ID!
-      name: String!
-      link: String!
-      type: String!
-      startDate: Date!
-      endDate: Date!
-    ): Advertisement!
+      input: CreateAdvertisementInput!
+    ): CreateAdvertisementPayload @auth
 
     createPost(data: PostInput!, file: String): Post @auth
 
     createUserTag(input: CreateUserTagInput!): UserTag @auth
 
     createSampleOrganization: Boolean! @auth
 
-    deleteAdvertisementById(id: ID!): DeletePayload!
+    deleteAdvertisement(id: ID!): DeletePayload!
```

Comment:
```graphql
type DeleteAdvertisementPayload {
  advertisement: Advertisement
}

deleteAdvertisement(id: ID!): DeleteAdvertisementPayload
```

Reply:
@xoldyckk actually the code is just returning the success message. So the current payload is fine.
```    // Deletes the ad.
    const deleteAdvertisement = await Advertisement.deleteOne({
      _id: args.id,
    });
    // Returns deleted ad.
    return { success: deleteAdvertisement.deletedCount ? true : false };
```


https://github.com/disha1202/talawa-api/blob/43f9ad971a42d3333bf086e568adbc4260a77d6e/src/resolvers/Mutation/deleteAdvertisement.ts#L45-L50

Reply:
I updated the code to return the deleted advertisement.